src/conversions.jl

```diff
@@ -591,6 +591,43 @@ function cal2jd(iy, im, id)
     jd, jd1
 end
 
+function dtf2d(scale, iy, im, id, ihr, imn, sec)
+    jd, w = cal2jd(iy, im, id)
+    jd += w
+    seclim = 60.0
+
+    if scale == "UTC"
+        dat0 = leapseconds(jd)
+        dat12 = leapseconds(jd + 0.5)
+        dat24 = leapseconds(jd + 1.5)
+
+        dleap = dat24 - (2.0 * dat12 - dat0)
+        adjusted_seconds_per_day = SECONDS_PER_DAY + dleap
+
+        if ihr == 23 && imn == 59
+            seclim += dleap
+        end
+    end
+
+    if ihr >= 0 && ihr <= 23
```

Comment:
Since we do not need to compute error codes like the C code, we do not need this convoluted structure. I would propose to move the argument checking to the top of the function and have separate clauses for each condition.